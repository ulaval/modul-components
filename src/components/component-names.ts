export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
