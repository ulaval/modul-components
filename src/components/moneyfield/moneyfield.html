<div class="m-moneyfield"
     :style="{ width: inputWidth, maxWidth: inputMaxWidth }">
    <m-decimalfield v-bind="bindData"
                    v-on="$listeners"
                    :force-rounding-format="true"
                    :precision="precision">
        <!-- content should stay inline to avoid adding useless white space -->
        <template v-if="currencyDetail.symbolPosition === 'before'"
                  slot="prefix">
            <span class="m-moneyfield__currency-symbol">
                {{ currencyDetail.localizedSymbol }}
            </span>
        </template>
        <template v-if="currencyDetail.symbolPosition === 'after'"
                  slot="suffix">
            <span class="m-moneyfield__currency-symbol">
                {{ currencyDetail.localizedSymbol }}
            </span>
        </template>
    </m-decimalfield>
</div>
