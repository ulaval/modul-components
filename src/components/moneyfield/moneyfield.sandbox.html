<div>
    <h2>Base moneyfield</h2>
    <m-moneyfield v-model="model"></m-moneyfield>
    <div>
        model value: {{ model }} is number = {{  isNumber }}
    </div>

    <h3 class="m-u--h6">Label</h3>
    <m-moneyfield label="label"></m-moneyfield>

    <h3 class="m-u--h6">Placeholder</h3>
    <m-moneyfield placeholder="write something clever"></m-moneyfield>

    <h3 class="m-u--h6">Required</h3>
    <m-moneyfield :required-marker="true"
                  label="Amount"
                  value=""></m-moneyfield>

    <h3 class="m-u--h6">Waiting</h3>
    <m-moneyfield :waiting="true"
                  label="label"></m-moneyfield>

    <h3 class="m-u--h6">Disabled</h3>
    <m-moneyfield :disabled="true"
                  label="label"></m-moneyfield>
    <m-moneyfield :disabled="true"
                  label="label"
                  value="value"></m-moneyfield>

    <h3 class="m-u--h6">Read only</h3>
    <m-moneyfield :readonly="true"
                  label="label"></m-moneyfield>
    <m-moneyfield :readonly="true"
                  label="label"
                  value="value"></m-moneyfield>
    <h2>Messages</h2>
    <h3 class="m-u--h6">Static error (default)</h3>
    <m-moneyfield label="label"
                  value="wrong answer"
                  error-message="this field is in error">
    </m-moneyfield>

    <h3 class="m-u--h6">Helper message</h3>
    <m-moneyfield label="label"
                  helper-message="this message is helping you understand the component awesome feature">
    </m-moneyfield>

    <h3 class="m-u--h6">Valid message</h3>
    <m-moneyfield label="label"
                  valid-message="Valid textfield"></m-moneyfield>
    <h2>Sizes</h2>
    <m-moneyfield label="x-small"
                  max-width="x-small"
                  value=""></m-moneyfield>
    <br />
    <m-moneyfield label="regular"
                  max-width="regular"
                  value=""></m-moneyfield>
    <br />
    <m-moneyfield label="medium"
                  max-width="medium"
                  value=""></m-moneyfield>
    <br />
    <m-moneyfield label="large"
                  max-width="large"
                  value=""></m-moneyfield>

    <p class="m-u--margin-top"><b>max-width</b> should be set at <b>none</b> to override default settings</p>
    <m-moneyfield label="custom width"
                  width="333px"
                  max-width="none"
                  value=""></m-moneyfield>
    <h2>Tag style</h2>
    <m-moneyfield label="h1"
                  tag-style="h1">
    </m-moneyfield>
    <br />
    <m-moneyfield label="h2"
                  tag-style="h2">
    </m-moneyfield>
    <br />
    <m-moneyfield label="h3"
                  tag-style="h3">
    </m-moneyfield>
    <br />
    <m-moneyfield label="h4"
                  tag-style="h4">
    </m-moneyfield>
    <br />
    <m-moneyfield label="h5"
                  tag-style="h5">
    </m-moneyfield>
    <br />
    <m-moneyfield label="h6"
                  tag-style="h6">
    </m-moneyfield>
    <br />
    <m-moneyfield label="p"
                  tag-style="p">
    </m-moneyfield>
    <h2>Localization</h2>
    <m-dropdown label="Language"
                placeholder="Choose a language"
                v-model="selectedLanguage">
        <m-dropdown-item v-for="language in languages"
                         :key="language"
                         :value="language"
                         :label="language"></m-dropdown-item>
    </m-dropdown>
</div>
