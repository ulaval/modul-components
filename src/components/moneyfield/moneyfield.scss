@import 'commons';

.m-moneyfield {
    display: inline-flex;

    @include m-input-inline-spacing();
}
