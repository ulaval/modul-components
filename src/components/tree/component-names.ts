export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
