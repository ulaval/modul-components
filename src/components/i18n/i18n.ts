import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';

import { ModulVue } from '../../utils/vue/vue';
import { I18N_NAME } from '../component-names';
import WithRender from './i18n.html';

@WithRender
@Component
export class MI18n extends ModulVue {
    @Prop()
    public k: string;
    @Prop({ default: () => [] })
    public params: any;
    @Prop()
    public nb?: number;
    @Prop()
    public modifier: string;
    @Prop({ default: true })
    public htmlEncode: boolean;

    protected created(): void {
        if (!this.$i18n) {
            throw new Error('<' + I18N_NAME + '>: this.$i18n is undefined, you must register the i18n plugin.');
        }
    }

    private get text(): string {
        let result = '';
        if (this.k) {
            result = this.$i18n.translate(
                this.k,
                this.params,
                this.nb,
                this.modifier,
                this.htmlEncode
            );
        }

        return result;
    }
}

const I18nPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(I18N_NAME, 'plugin.install');
        v.component(I18N_NAME, MI18n);
    }
};

export default I18nPlugin;
