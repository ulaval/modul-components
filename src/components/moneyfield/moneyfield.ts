import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import { InputLabel } from '../../mixins/input-label/input-label';
import { InputManagement } from '../../mixins/input-management/input-management';
import { InputState } from '../../mixins/input-state/input-state';
import { InputMaxWidth, InputWidth } from '../../mixins/input-width/input-width';
import L10nPlugin, { MCurrency } from '../../utils/l10n/l10n';
import { ModulVue } from '../../utils/vue/vue';
import { MONEYFIELD_NAME } from '../component-names';
import DecimalfieldPlugin from '../decimalfield/decimalfield';
import MessagePlugin from '../message/message';
import WithRender from './moneyfield.html?style=./moneyfield.scss';


// TODO: Everything here is dispensable.  This will be all replaced by a wrapper of decimal-field.
@WithRender
@Component({
    inheritAttrs: false,
    mixins: [
        InputState,
        InputWidth,
        InputLabel,
        InputManagement
    ]
})
export class MMoneyfield extends ModulVue {
    @Prop()
    value: number;

    @Prop({ default: InputMaxWidth.Small })
    public maxWidth: string;

    get currencyDetail(): MCurrency {
        return (this as ModulVue).$l10n.getCurrencyDetail(this.currentLocale);
    }

    get currentLocale(): string {
        return (this as ModulVue).$i18n.currentLocale;
    }

    get bindData(): any {
        return Object.assign({}, this.$props || {}, this.$attrs || {});
    }
}

const MoneyFieldPlugin: PluginObject<any> = {
    install(v): void {
        v.use(MessagePlugin);
        v.use(L10nPlugin);
        v.use(DecimalfieldPlugin);
        v.component(MONEYFIELD_NAME, MMoneyfield);
    }
};

export default MoneyFieldPlugin;
