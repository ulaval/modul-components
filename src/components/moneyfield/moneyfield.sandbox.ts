import Vue, { PluginObject } from 'vue';
import { Component, Watch } from 'vue-property-decorator';
import { ENGLISH, FRENCH, Messages } from '../../utils/i18n/i18n';
import { ModulVue } from '../../utils/vue/vue';
import { MONEYFIELD_NAME } from '../component-names';
import MoneyFieldPlugin from './moneyfield';
import WithRender from './moneyfield.sandbox.html';


@WithRender
@Component
export class MMoneyfieldSandbox extends Vue {
    languages: string[] = [FRENCH, ENGLISH];
    selectedLanguage: string = (Vue.prototype as ModulVue).$i18n.currentLang();
    originalLang = (Vue.prototype as ModulVue).$i18n.currentLang();
    i18n: Messages = Vue.prototype.$i18n;
    model: number = 0;

    get isNumber(): boolean {
        return (typeof this.model === 'number');
    }

    @Watch('selectedLanguage')
    changeLanguage(): void {
        this.i18n.currentLang(this.selectedLanguage);
    }

    destroyed(): void {
        this.i18n.currentLang(this.originalLang);
    }
}

const MMoneyfieldSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.use(MoneyFieldPlugin);
        v.component(`${MONEYFIELD_NAME}-sandbox`, MMoneyfieldSandbox);
    }
};

export default MMoneyfieldSandboxPlugin;
