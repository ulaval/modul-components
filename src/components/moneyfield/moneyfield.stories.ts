import { withA11y } from '@storybook/addon-a11y';
import { select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ENGLISH, FRENCH } from '../../utils/i18n/i18n';
import { ModulVue } from '../../utils/vue/vue';
import { MONEYFIELD_NAME } from './../component-names';
import { MMoneyfield } from './moneyfield';

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Basic', () => ({
        components: { MMoneyfield },
        template: `
            <div>
                <div><${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :label="'Label'"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :label="'Required label'" :required-marker="true"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :valid-message="'Valid message'"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :helper-message="'Helper message'"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :error-message="'Error message'"></${MONEYFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('States', () => ({
        components: { MMoneyfield },
        template: `
            <div>
                <div><${MONEYFIELD_NAME} v-model="value" :disabled="true" :placeholder="'Disabled'"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :placeholder="'Read-only'" :readonly="true"></${MONEYFIELD_NAME}></div>
                <div><${MONEYFIELD_NAME} v-model="value" :placeholder="'Waiting'" :waiting="true"></${MONEYFIELD_NAME}></div>
            </div>
        `,
        data: () => ({
            value: undefined
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('With initial value (0)', () => ({
        components: { MMoneyfield },
        template: `<${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}>`,
        data: () => ({
            value: 0
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('With initial value', () => ({
        components: { MMoneyfield },
        template: `<${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}>`,
        data: () => ({
            value: 123456.7
        })
    }));

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Localization', () => ({
        components: { MMoneyfield },
        template: `
            <${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}>
        `,
        props: {
            lang: {
                default: select('Language', {
                    [FRENCH]: FRENCH,
                    [ENGLISH]: ENGLISH
                }, (Vue.prototype as ModulVue).$i18n.currentLang())
            }
        },
        watch: {
            lang: function(): void {
                this.i18n.currentLang(this.lang);
            }
        },
        data: () => ({
            value: 123456.7,
            originalLang: (Vue.prototype as ModulVue).$i18n.currentLang(),
            i18n: Vue.prototype.$i18n
        }),
        destroyed(): void {
            (Vue.prototype as ModulVue).$i18n.currentLang(this.originalLang);
        }
    }));

// storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
//     .addDecorator(withA11y)
//     .addDecorator(withKnobs)
//     .add('Format', () => ({
//         components: { MMoneyfield },
//         template: `<div>
//                 <${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}>
//                 => {{ formatedOuput }}
//             </div>
//         `,
//         props: {
//             lang: {
//                 default: select('Language', {
//                     [FRENCH]: FRENCH,
//                     [ENGLISH]: ENGLISH
//                 }, (Vue.prototype as ModulVue).$i18n.currentLang())
//             },
//             stripDecimalZeroes: {
//                 default: boolean('Strip decimal zeroes', false)
//             }
//         },
//         watch: {
//             lang: function(): void {
//                 this.i18n.currentLang(this.lang);
//             }
//         },
//         computed: {
//             formatedOuput(): string {
//                 return formatCurrencyWithOptions(this.value, {
//                     currency: (Vue.prototype as ModulVue).$l10n.currentCurrency,
//                     stripDecimalZeroes: this.stripDecimalZeroes
//                 });
//             }
//         },
//         data: () => ({
//             value: 123456.7,
//             originalLang: (Vue.prototype as ModulVue).$i18n.currentLang(),
//             i18n: Vue.prototype.$i18n,
//             l10n: Vue.prototype.$l10n
//         }),
//         destroyed(): void {
//             (Vue.prototype as ModulVue).$i18n.currentLang(this.originalLang);
//         }
//     }));

storiesOf(`${componentsHierarchyRootSeparator}${MONEYFIELD_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('Default precision', () => ({
        components: { MMoneyfield },
        template: `
            <${MONEYFIELD_NAME} v-model="value"></${MONEYFIELD_NAME}>
        `,
        data: () => ({
            value: 999999999.99
        })
    }))
    .add('Custom precision', () => ({
        components: { MMoneyfield },
        template: `
            <${MONEYFIELD_NAME} v-model="value" :precision="7"></${MONEYFIELD_NAME}>
        `,
        data: () => ({
            value: 99999.99
        })
    }));
