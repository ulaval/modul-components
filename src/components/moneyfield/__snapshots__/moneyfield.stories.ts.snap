// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-moneyfield Basic 1`] = `
<div>
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        maxwidth="small"
        precision="11"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        label="Label"
        maxwidth="small"
        precision="11"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        label="Required label"
        maxwidth="small"
        precision="11"
        requiredmarker="true"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        maxwidth="small"
        precision="11"
        tagstyle="default"
        validmessage="Valid message"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        helpermessage="Helper message"
        maxwidth="small"
        precision="11"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        errormessage="Error message"
        maxwidth="small"
        precision="11"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-moneyfield Custom precision 1`] = `
<div
  class="m-moneyfield"
  style="width: 100%; max-width: 136px;"
>
  <m-decimalfield
    maxwidth="small"
    precision="7"
    tagstyle="default"
    value="99999.99"
    width="100%"
  >
    <template>
      CA$
    </template>
     
    <!---->
  </m-decimalfield>
</div>
`;

exports[`Storyshots components|m-moneyfield Default precision 1`] = `
<div
  class="m-moneyfield"
  style="width: 100%; max-width: 136px;"
>
  <m-decimalfield
    maxwidth="small"
    precision="11"
    tagstyle="default"
    value="999999999.99"
    width="100%"
  >
    <template>
      CA$
    </template>
     
    <!---->
  </m-decimalfield>
</div>
`;

exports[`Storyshots components|m-moneyfield Localization 1`] = `
<div
  class="m-moneyfield"
  style="width: 100%; max-width: 136px;"
>
  <m-decimalfield
    maxwidth="small"
    precision="11"
    tagstyle="default"
    value="123456.78"
    width="100%"
  >
    <template>
      CA$
    </template>
     
    <!---->
  </m-decimalfield>
</div>
`;

exports[`Storyshots components|m-moneyfield States 1`] = `
<div>
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        disabled="true"
        maxwidth="small"
        placeholder="Disabled"
        precision="11"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        maxwidth="small"
        placeholder="Read-only"
        precision="11"
        readonly="true"
        tagstyle="default"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
   
  <div>
    <div
      class="m-moneyfield"
      style="width: 100%; max-width: 136px;"
    >
      <m-decimalfield
        maxwidth="small"
        placeholder="Waiting"
        precision="11"
        tagstyle="default"
        waiting="true"
        width="100%"
      >
        <template>
          CA$
        </template>
         
        <!---->
      </m-decimalfield>
    </div>
  </div>
</div>
`;

exports[`Storyshots components|m-moneyfield With initial value (0) 1`] = `
<div
  class="m-moneyfield"
  style="width: 100%; max-width: 136px;"
>
  <m-decimalfield
    maxwidth="small"
    precision="11"
    tagstyle="default"
    value="0"
    width="100%"
  >
    <template>
      CA$
    </template>
     
    <!---->
  </m-decimalfield>
</div>
`;

exports[`Storyshots components|m-moneyfield With initial value 1`] = `
<div
  class="m-moneyfield"
  style="width: 100%; max-width: 136px;"
>
  <m-decimalfield
    maxwidth="small"
    precision="11"
    tagstyle="default"
    value="123456.78"
    width="100%"
  >
    <template>
      CA$
    </template>
     
    <!---->
  </m-decimalfield>
</div>
`;
